service/history/workflowTaskHandlerCallbacks.go

```diff
@@ -507,6 +513,8 @@ func (handler *workflowTaskHandlerCallbacksImpl) handleWorkflowTaskCompleted(
 			request.GetIdentity(),
 			completedEvent.GetEventId(), // If completedEvent is nil, then GetEventId() returns 0 and this value shouldn't be used in workflowTaskHandler.
 			ms,
+			weContext.UpdateRegistry(ctx),
+			&effects,
```

Comment:
nit: the function `newWorkflowTaskHandler` has too many arguments.

Reply:
I don't disagree. But that's going to be out of scope for this PR

Reply:
https://www.brainyquote.com/quotes/alan_perlis_177279 :smile:

Reply:
This is genius. I haven't heard it!